### PR TITLE
Feature/socks5 crate

### LINKED ISSUE #3201: [helios2] Factor out socks5-client from NymConnect into crate to be used from the SDK and NymConnect (avoid using `NymConfig` in SDK)

### PR DESCRIPTION
**Pull request by neacsu**
# Description

Closes: #3201

# Checklist:

- [x] added a changelog entry to `CHANGELOG.md`

### CODE REVIEW

File: clients/socks5/src/lib.rs
Comment (octol) [inline clients/socks5/src/lib.rs:1]: Thinking we can probably get rid of this entirely now? The lib isn't used anywhere anymore? And remove the lib part in the Cargo.toml file.

File: common/client-core/src/config/persistence/key_pathfinder.rs [hunk 1]
```diff
@@ -18,7 +18,10 @@ pub struct ClientKeyPathfinder {
 impl ClientKeyPathfinder {
     pub fn new(id: String) -> Self {
         let os_config_dir = dirs::config_dir().expect("no config directory known for this OS"); // grabs the OS default config dir
-        let config_dir = os_config_dir.join("nym").join("clients").join(id);
+        let config_dir = os_config_dir
+            .join("nym")
+            .join("../../../../../clients")
```
Comment (octol) [inline common/client-core/src/config/persistence/key_pathfinder.rs]: This line accidentally left in?

File: sdk/rust/nym-sdk/examples/socks5.rs [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+    println!("Connecting sender");
+    let mut sending_client = sending_client.connect_to_mixnet().await.unwrap();
+    // wait until socks5 server is started
+    println!("Wait 5 seconds for the socks5 setup to be done");
+    std::thread::sleep(std::time::Duration::from_secs(5));
+
+    let proxy = reqwest::Proxy::all(format!(
+        "socks5h://127.0.0.1:{}",
+        nym_network_defaults::DEFAULT_SOCKS5_LISTENING_PORT
```
Comment (octol) [inline sdk/rust/nym-sdk/examples/socks5.rs:28]: Would be neat if you could get the port from querying the sending client instance

**Review by octol — APPROVED** [review]

File: sdk/rust/nym-sdk/examples/socks5.rs (hunk 2 above)
Comment (neacsu) [reply sdk/rust/nym-sdk/examples/socks5.rs:28]: I will do that as part of https://github.com/nymtech/nym/issues/3237 , and possibly expose other socks5-specific structures.

**Pull request merged**